src/controller.js

```diff
@@ -120,14 +119,24 @@ export class Controller {
       observer.selfSubscriber = selfSubscriber;
     }
 
+    let overrideContext;
     if (this.view !== null) {
       if (skipSelfSubscriber) {
         this.view.viewModelScope = scope;
       }
-
-      this.view.bind(this.viewModel, createOverrideContext(this.viewModel, scope.overrideContext));
+      if (this.viewModel === scope.overrideContext.bindingContext) {
```

Comment:
This fixes a bug where an `overrideContext` was becoming it's own `parentOverrideContext`.